wasm.md

```diff
@@ -636,38 +676,70 @@ Currently, in the expanded form, the `export`s will come after the definition of
     syntax FuncExports ::= List{FuncExport, ""} [klabel(listFuncExport)]
 ```
 
+### Function Local Declaration
+
+```k
+    syntax LocalDecl      ::= "(" LocalDecl ")"     [bracket]
+                            | "local" ValTypes
+    syntax LocalDecls     ::= List{LocalDecl , ""} [klabel(listLocalDecl)]
+ // ----------------------------------------------------------------------
+
+    syntax VecType ::=  asLocalType ( LocalDecls            ) [function]
+                     | #asLocalType ( LocalDecls , ValTypes ) [function]
+ // -------------------------------------------------------------------
+    rule  asLocalType(LDECLS) => #asLocalType(LDECLS, .ValTypes)
+
+    rule #asLocalType(.LocalDecls            , VTYPES) => [ VTYPES ]
+    rule #asLocalType(local VTYPES'   LDECLS , VTYPES)
+      => #asLocalType(                LDECLS , VTYPES + VTYPES')
+```
+
+### Function Implicit Type Declaration
+
+It could also be declared implicitly when a `TypeUse` is a `TypeDecls`, in this case it will allocate a type when the type is not in the current module instance.
+
+```k
+    syntax Instr ::= #checkTypeUse ( TypeUse )
+ // ------------------------------------------
+    rule <k> #checkTypeUse ( TDECLS:TypeDecls ) => (type (func TDECLS)) ... </k>
+         <types> TYPES </types> requires #reverseLookup ( TYPES , asFuncType(TDECLS) ) ==Int -1
+    rule <k> #checkTypeUse ( _ ) => . ... </k> [owise]
```

Comment:
We can make this a bit better (avoiding `owise` and removing `#reverseLookup`):

```
    rule <k> #checkTypeUse ( TDECLS:TypeDecls ) => (type (func TDECLS)) ... </k>
         <types> TYPES </types> requires asFuncType(TDECLS) in values(TYPES)
    rule <k> #checkTypeUse ( TDECLS ) => . ... </k>
         <types> TYPES </types> requires notBool asFuncType(TDECLS) in values(TYPES) 
```

Reply:
Also, since it's probably not the case that `<types>` will every be symbolic, you can do:

```
rule <k> #checkTypeUse ( TDECLS:TypeDecls )
       => #if notBool asFuncType(TDECLS) in values(TYPES)
                #then (type (func TDECLS))
                #else .K
             #fi
          ...
          </k>
         <types> TYPES </types>
```